config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java

```diff
@@ -32,59 +41,155 @@
+                                                                             final String projectName, final String projectVersion, final String uniqueName) {
+        Element policyElement = (Element) entityXml.cloneNode(true);
+        final Element policyDetails = getSingleChildElement(policyElement, POLICY_DETAIL);
+        Element nameElement = getSingleChildElement(policyDetails, NAME);
+        nameElement.setTextContent(uniqueName);
+        final Element resources = getSingleChildElement(policyElement, RESOURCES);
+        final Element resourceSet = getSingleChildElement(resources, RESOURCE_SET);
+        final Element resource = getSingleChildElement(resourceSet, RESOURCE);
+        final String policyString = resource.getTextContent();
+        Document policyDocument;
+        try {
+            policyDocument = stringToXMLDocument(documentTools, policyString);
+        } catch (DocumentParseException e) {
+            throw new EntityBuilderException("Could not load policy: " + e.getMessage(), e);
+        }
+        if (policyDocument != null) {
+            Element policy = policyDocument.getDocumentElement();
+            NodeList assertionReferences = policy.getElementsByTagName(PolicyXMLElements.ENCAPSULATED);
+            if (assertionReferences != null && assertionReferences.getLength() > 0) {
+                for (int i = 0; i < assertionReferences.getLength(); i++) {
+                    Node assertionElement = assertionReferences.item(i);
+                    if (!(assertionElement instanceof Element)) {
+                        throw new EntityBuilderException("Unexpected assertion node type: " + assertionElement.getNodeName());
                     }
-                });
+                    Element encassElement = (Element) assertionElement;
+                    Element encassConfigElement = getSingleChildElement(encassElement, PolicyXMLElements.ENCAPSULATED_ASSERTION_CONFIG_NAME);
+                    final Node encassNameNode = encassConfigElement.getAttributeNode(ATTRIBUTE_STRING_VALUE);
+                    final String encassName = encassNameNode.getNodeValue();
+                    final Encass encassEntity = encassMap.get(encassName);
+                    Set<Annotation> annotations = encassEntity != null ? encassEntity.getAnnotations() : null;
+                    if (annotations == null || !(annotations.stream().anyMatch(annotation -> ANNOTATION_TYPE_REUSABLE_ENTITY.equals(annotation.getType())))) {
+                        encassNameNode.setNodeValue(getUniqueName(projectName, projectVersion, annotatedEntity, encassName));
+                    }
+                }
+                resource.setTextContent(documentTools.elementToString(policy));
+            }
+        }
+        return policyElement;
+    }
+
+    private String getUniqueName(final String projectName, final String projectVersion, final AnnotatedEntity annotatedEntity,
```

Comment:
Can you provide an example in the comment, what this returned value will look like?

Reply:
done